remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java

```diff
@@ -188,24 +195,19 @@ public CollectionWithPagingInfo<Tag> getTags(StoreRef storeRef, Parameters param
 
     public NodeRef validateTag(String tagId)
     {
-    	NodeRef tagNodeRef = nodes.validateNode(tagId);
-    	if(tagNodeRef == null)
-    	{
-    		throw new EntityNotFoundException(tagId);
-    	}
-    	return tagNodeRef;
+        NodeRef tagNodeRef = nodes.validateNode(tagId);
+        return validateTag(tagNodeRef.getStoreRef(), tagId);
     }
     
     public NodeRef validateTag(StoreRef storeRef, String tagId)
     {
-    	NodeRef tagNodeRef = nodes.validateNode(storeRef, tagId);
-    	if(tagNodeRef == null)
-    	{
-    		throw new EntityNotFoundException(tagId);
-    	}
-    	return tagNodeRef;
+        NodeRef tagNodeRef = nodes.validateNode(storeRef,tagId);
```

Comment:
This looks wrong to me. In both places that you've added a call to this method we're now calling:
```
NodeRef tagNodeRef = nodes.validateNode(tagId)
NodeRef tagNodeRef = nodes.validateNode(storeRef,tagId);
```
(The first call outside this method and second call inside this method).

Please could you add some justification if you think this is correct?

Reply:
Yes in current implementation we are calling Node validation method twice, So we have now completely independent
implementation for both the methods .